src/aind_data_transfer/jobs/zarr_upload_job.py

```diff
@@ -337,8 +359,44 @@ def run_job(self):
         # Exclude raw image directory, this is uploaded separately
         self._upload_to_s3(
             dir=self._data_src_dir,
-            excluded=os.path.join(self._raw_image_dir, "*"),
-        )
+            excluded=os.path.join(self._raw_image_dir, "*"), )
+
+
+def _cleanup(deployment: str) -> None:
+    """
+    Clean up any resources that were created during the job.
+
+    Parameters
+    ----------
+    deployment : str
+      The type of deployment. Either "local" or "slurm"
+    """
+    if deployment == Deployment.SLURM.value:
+        job_id = os.getenv("SLURM_JOBID")
+        if job_id is not None:
+            try:
+                api_url = f"http://{os.environ['HPC_HOST']}"
+                api_url += f":{os.environ['HPC_PORT']}"
+                api_url += f"/{os.environ['HPC_API_ENDPOINT']}"
+                headers = {
+                    "X-SLURM-USER-NAME": os.environ["HPC_USERNAME"],
+                    "X-SLURM-USER-PASSWORD": os.environ["HPC_PASSWORD"],
+                    "X-SLURM-USER-TOKEN": os.environ["HPC_TOKEN"],
+                }
+            except KeyError as ke:
+                logging.error(
+                    f"Failed to get SLURM environment variables for cleanup: {ke} "
+                )
+                return
+            logging.info(f"Cancelling SLURM job {job_id}")
+            response = cancel_slurm_job(job_id, api_url, headers)
```

Comment:
It does make me slightly nervous about making requests to the HPC from within jobs running on the HPC. If this is unavoidable due to the dask bug, so be it, but should consider an alternative solution in the future.

Reply:
I agree it isn't ideal, but it seemed like the least risky way of triggering `scancel` once the job is done, since alternatives would require some type of communication between an external watchdog of some kind with the compute process, which would watch for some event (e.g., writing a specific file after the upload is complete). Unfortunately, calling `sys.exit()` from within the script is insufficient to kill the dask-mpi cluster, which will run idly until the job time limit.

Reply:
I think this should be fine in the short term. Long-term, having a watchdog is probably the way to go. I'll let Camilo and Carson give your PR a look.

Reply:
Totally agreed with Jon on this one.